lib/test_ids.rb

```
@@ -19,42 +19,92 @@ module TestIds
+      
+      # Return existing or create new store object in store hash
+      @store[config_id] ||= Store.new(config_id)
     end
 
     def allocator
       unless @configuration
         fail 'The test ID generator has to be configured before you can start using it'
       end
-      @allocator ||= Allocator.new
+
+      # Note: TestIds will keep a hash of allocators
+      #
+      @allocator = {} if @allocator.nil?       # Empty has if this is first allocator object
+
+      # Return existing or create new store object in allocator hash
+      @allocator[config_id] ||= Allocator.new(config_id)     
     end
 
+    # Returns the id of the current configuration in-use, use this to access store and allocator
+    def config_id
+      current_configuration.id
+    end
+    
     def configuration
       if block_given?
         configure do |config|
           yield config
         end
       else
-        @configuration ||
+        @configuration[config_id] ||
           fail('You have to create the configuration first before you can access it')
       end
     end
     alias_method :config, :configuration
 
-    def configure
-      if @configuration
-        fail "You can't modify an existing test IDs configuration"
+    def configure(options = {})
+      # Don't force 'id' usage
+      _id = options[:id] || :not_specified
+
+      # Create empty hash if first configuration
+      @configuration = {} if @configuration.nil?
+
+      # Note: TestIds will keep a hash of configurations with the 'id' as the key
+      #
+      # If this configuration doesn't exist, create new configuration (hash item)
+      # Else, shoot user a warning and set configuration as current (but don't allow bin/tnum changes)
+      if @configuration[_id].nil?
+        @configuration[_id] ||= Configuration.new(_id)
+        set_current_configuration(_id)
+      else
+        Origen.log.warn "Warning: You are attempting to configure an existing configuration: '#{_id}'"
```

I don't think we should warn in this case, it is going to be hit regularly in a std integration and don't want the warning to make the app owner try and add logic to prevent hitting it. In fact, since the hit is required to set the current config ID that could lead to more problems.

Longer term, we could pass back a dummy config object here, then compare it to the existing when we get it back and raise an error rather than a warning if it is different. I don't think that is a nice to have though and not required.

---

Done.